services/motion/builtin/builtin_test.go

```diff
@@ -1400,6 +1401,170 @@ func TestMoveOnGlobe(t *testing.T) {
+		// this is the case where we want to use check plan to show that a new obstacle causes collision
+		// define static obstacle
+		obstacleLeft, err := spatialmath.NewBox(
+			spatialmath.NewPose(r3.Vector{0.22981e3, -0.38875e3, 0},
+				&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 45}),
+			r3.Vector{900, 10, 10},
+			"obstacleLeft",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		req.Obstacles = []spatialmath.Geometry{obstacleLeft}
+
+		// construct move request
+		planExecutor, err := ms.(*builtIn).newMoveOnMapRequest(ctx, req, nil, 0)
+		test.That(t, err, test.ShouldBeNil)
+		mr, ok := planExecutor.(*moveRequest)
+		test.That(t, ok, test.ShouldBeTrue)
+
+		// construct plan
+		planResp, err := mr.Plan(ctx)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(planResp.Waypoints), test.ShouldBeGreaterThan, 2)
+
+		// place obstacle in opposte position and show that the generate path
+		// collides with obstacleRight
+		obstacleRight, err := spatialmath.NewBox(
+			spatialmath.NewPose(r3.Vector{0.89627e3, -0.37192e3, 0},
+				&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: -45}),
+			r3.Vector{900, 10, 10},
+			"obstacleLeft",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		wrldSt, err := referenceframe.NewWorldState(
+			[]*referenceframe.GeometriesInFrame{
+				referenceframe.NewGeometriesInFrame(
+					referenceframe.World,
+					[]spatialmath.Geometry{obstacleRight},
+				),
+			}, nil,
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = motionplan.CheckPlan(
+			mr.planRequest.Frame,
+			planResp.Motionplan,
+			wrldSt,
+			mr.planRequest.FrameSystem,
+			spatialmath.NewPose(
+				r3.Vector{X: 0.58772e3, Y: -0.80826e3, Z: 0},
+				&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 0},
+			),
+			referenceframe.FloatsToInputs([]float64{0, 0, 0}),
+			spatialmath.NewZeroPose(),
+			lookAheadDistanceMM,
+			logger,
+		)
+		test.That(t, err, test.ShouldNotBeNil)
+	})
+
+	t.Run("fail due to obstacles enclosing goals", func(t *testing.T) {
+		// define static obstacles
+		obstacleTop, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.64603e3, 0.77151e3, 0}),
+			r3.Vector{400, 10, 10},
+			"obstacleTop",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		obstacleBottom, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.64603e3, 0.42479e3, 0}),
+			r3.Vector{400, 10, 10},
+			"obstacleBottom",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		obstacleLeft, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.47525e3, 0.65091e3, 0}),
+			r3.Vector{10, 400, 10},
+			"obstacleLeft",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		obstacleRight, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.82183e3, 0.64589e3, 0}),
+			r3.Vector{10, 400, 10},
+			"obstacleRight",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		req.Obstacles = []spatialmath.Geometry{obstacleTop, obstacleBottom, obstacleLeft, obstacleRight}
```

Comment:
Why use multiple obstacles here?  One at the goal location should be sufficient I think

Reply:
I specifically requested this :)

Having an obstacle overlap the goal location yields different failure modes than having the base be able to validly exist at the goal location, but be unable to connect to the start.